### PR TITLE
Add curve filter for applying RGB curves.

### PR DESCRIPTION
**Pull request by gockxml**
## Usage

    /filters:curve(<curve of all channels>, <curve of R>, <curve o f G>, <curve of B>)

### Format of each curve

    [(x1,y1),(x2,y2),...]

#### Original image
image

#### Demo 1
    /filters:curve([(0,0),(40,59),(255,255)],[(32,59),(64,80),(92,111),(128,153),(140,169),(164,201),(192,214),(224,215),(240,214),(255,212)],[(34,41),(64,51),(92,76),(128,112),(140,124),(164,147),(192,180),(224,216),(240,236),(255,255)],[(40,46),(64,55),(92,83),(128,127),(140,144),(164,174),(192,197),(224,199),(240,197),(255,198)])/
image

#### Demo 2
    /filters:curve([(0,0),(255,255)],[(0,50),(16,51),(32,69),(58,85),(92,120),(128,170),(140,186),(167,225),(192,245),(225,255),(244,255),(255,254)],[(0,0),(16,2),(32,18),(64,59),(92,116),(128,182),(167,211),(192,227),(224,240),(244,247),(255,252)],[(0,48),(16,50),(62,77),(92,110),(128,144),(140,153),(167,180),(192,192),(224,217),(244,225),(255,225)])/
image

### CODE REVIEW

**Comment by heynemann** [comment]
Sorry @gockxml, I was about to merge your PR, but github says I can't merge until it's rebased. Can you do it? Thanks!

**Comment by gockxml** [comment]
No problem. Rebased.

File: thumbor/config.py [hunk 1]
```diff
@@ -267,6 +267,7 @@
         'thumbor.filters.no_upscale',
         'thumbor.filters.saturation',
         'thumbor.filters.max_age',
+        'thumbor.filters.curve'
```
Comment (crccheck) [inline thumbor/config.py]: I think there should be a trailing comma here so future changes won't cause a diff on this line.

Comment (gockxml) [inline thumbor/config.py]: My bad. Thank you!

**Comment by heynemann** [comment]
Sorry but when applying this PR, the build fails due to a test that verifies all the default filters. Can you fix it? The log for the test failure is here:

https://api.travis-ci.org/jobs/46025767/log.txt?deansi=true

And the travis build is here:

https://travis-ci.org/thumbor/thumbor/builds/46025766

Thanks!!!

**Comment by gockxml** [comment]
@heynemann thank you very much for showing me the complete log. For some weird reason, I couldn't see the whole. I thought the building failure occurred during environment setup. Sorry for my clumsy.

**Comment by heynemann** [comment]
Thanks a LOT for your contribution.

**Comment by heynemann** [comment]
Hey @gockxml 

Can you please create a wiki page for this plug-in?

Thanks! We'll release a version now!

**Comment by gockxml** [comment]
Added at https://github.com/thumbor/thumbor/wiki/Filters. Thank you!

**Comment by heynemann** [comment]
Brilliant!

**Pull request merged**